### ISSUE TITLE
QuickPickItem should allow extensions to specify accessibilityInformation field 

### ISSUE THREAD

**Issue by pragyamehta** [microsoft/vscode#133169]
Learn more should have a role `link`

![image](https://user-images.githubusercontent.com/1926584/134644235-f346e210-504e-4991-b38b-5894b300b973.png)

**Comment by isidorn** [comment]
@pragyamehta what control type do you recommend this item to have and why?

**Comment by isidorn** [comment]
@pragyamehta can you please attach the picture of the Learn More link that you are referring to. Thanks

**Comment by pragyamehta** [comment]
![image](https://user-images.githubusercontent.com/22248649/134393606-2a4dbda8-23a3-4a7e-8963-74892b7b53b8.png)

**Comment by isidorn** [comment]
@TylerLeonhardt this is a request that extensions should be able to specify `accessibilityInformation` for `QuickPickItem`.
This is already possible for [StatusBarItems](https://github.com/microsoft/vscode/blob/main/src/vs/vscode.d.ts#L5792) and [TreeItems](https://github.com/microsoft/vscode/blob/main/src/vs/vscode.d.ts#L9549).

I think it makes sense that we also add this to our `QuickPickItem` API. With that extensions could say that the role of the item is actually a `link` and we will be able to set a precise `role` on the element.

What do you think?

**Comment by TylerLeonhardt** [comment]
I think that's reasonable... I'm wondering if @chrmarti has any context on this. I know that there was another issue (https://github.com/microsoft/vscode/issues/120176#issuecomment-836974860) around properly setting the role for the checkboxes for multi-select and I'm wondering if that would also be fixed with this.

**Comment by isidorn** [comment]
@TylerLeonhardt I think for that issue you know on the client side that it is a checkbox (since you render a checkbox). So you can properly set the role.
In this case you on the client side do not know this is a link, right? And that is why an extension would have to contribute the role. Or in the case you know it is a link this makes the problem easier

Let's also see what @chrmarti says.
Thanks!

**Comment by chrmarti** [comment]
Following StatusBarItems and TreeItems makes sense. I don't remember any specifics here, it looks like this has been updated by others over time: https://github.com/microsoft/vscode/blame/main/src/vs/base/parts/quickinput/browser/quickInputList.ts#L734

**Comment by TylerLeonhardt** [comment]
> In this case you on the client side do not know this is a link, right? And that is why an extension would have to contribute the role. Or in the case you know it is a link this makes the problem easier

Yes that makes sense.

> it looks like this has been updated by others over time

Thanks for the pointer @chrmarti !

**Comment by isidorn** [comment]
@TylerLeonhardt is it possible to assign this for October or November, what do you think?

**Comment by TylerLeonhardt** [comment]
@isidorn Oct is debt month, yeah? Do we normally do these kinds of things in that month?

**Comment by isidorn** [comment]
@TylerLeonhardt depends on the person I guess.
Debt month I usually cleaned up my issues, tackled debt and tried to tackle low hanging fruit.
I always looked at debt like "get your github issues in order and fix what makes sense" :)

**Comment by TylerLeonhardt** [comment]
I'm not sure this is a low hanging fruit but I'll add it to Oct and see what happens.

**Comment by isidorn** [comment]
@TylerLeonhardt putting to November is perfectly fine as I initially suggested mentioned. Whatever works best for you. Thanks!

**Comment by fsteffi** [comment]
**GitHub Tags:**#A11yMAS;WCAG4.1.2;A11ySev2;#A11yTCS;

**Comment by fsteffi** [comment]
approved to closed by SME

**Comment by isidorn** [comment]
Thank you very much @fsteffi 
Based on recommendation from Accessibility team closing as out of scope.

More details:
Reason for severity 2 per definition here the user has to be blocked or unable to complete the scenario. However in this issue this is not the case, the only problem for the issue is that an item in the quick pick has a wrong role. The screen reader will still read the correct label and user can trigger that item, the only unexpected thing is that Screen Reader will not read that it is a link. Users expect to be able to click on quick pick items, so nothing is blocking here.

**Comment by isidorn** [comment]
Assigning to November so we have it as a reminder for next plan.

**Comment by fsteffi** [comment]
@isidorn  and @Matthew Fisher

this bug was closed approved by SME any reason to have this open

**Comment by isidorn** [comment]
@fsteffi I believe another bug depends on this issue. We would love to close it as out of scope if that is possible. Let me also check with Tanaz.

**Comment by MicroFish91** [comment]
Correct, the issue linked just above was dependent on this one.  If this one is approved to be closed out, could we please have ours revisited and possibly closed out as well?  Thank you :)

**Comment by isidorn** [comment]
Just got approval from Accessibility team that we can close this as approved by subject matter expert.
